simpleviewinc/keg-components
# WIP: Add svkeg repos

## Context

* Github NPM packages are a pain to work with
* They always require a git key
* They require a .npmrc to overwrite the default settings

## Goal

* Update the repo to be deployed to npm
  * https://www.npmjs.com/package/@svkeg/re-theme
* Use the `@svkeg` organization
* Updated the dependencies to use the `@svkeg` org
  * See https://www.npmjs.com/search?q=%40svkeg

## Updates

* `.npmrc`
  * Removed
* `package.json`
  * Updated to be deployed to npm
* Removed a number of alias
  * Updated web to use native components through react-native-web
  * This removes the need for those alias
  * In some cases the web-component was kept
* Updated / cleanup a number of components
* Cleaned up styles on switch and checkbox components
* Cleaned up form. input styles
  * Added a derived input style
    * Builds the style rules from defaults.json
* `...all other files`
  * Updated to use `@svkeg/jsutils`

## Testing

* Pull branch
* Install deps => `yarn install`
* Run tests => `yarn test`
  * Ensure all tests pass
* Run the docker packages => `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:add-svkeg-repos`
  * Ensure everything works as expected
